### PR TITLE
Create a MethodSummaryTable

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Adds a MethodSummaryTable. For now, this will be used in the instructions panel but may eventually be added to the programming class page. In the mock, part of the name was hyperlinked, which would've been difficult, so I added an icon link here. I'm not sure what the best icon would be so I'm open to suggestions!

Screen Shot 2022-05-12 at 3 33 40 PM

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/codeDocs/MethodSummaryTable.jsx
Comment (bethanyaconnor) [inline apps/src/templates/codeDocs/MethodSummaryTable.jsx:12]: I imagine that for the instructions panel we would want the link to open in a new tab but when this is added to the programming class page, we wouldn't. If it seems like overkill to add this prop now I can remove it.

Comment (molly-moen) [reply apps/src/templates/codeDocs/MethodSummaryTable.jsx:12]: I think it makes sense to add it now!

File: apps/src/templates/codeDocs/MethodSummaryTable.jsx
Comment (molly-moen) [inline apps/src/templates/codeDocs/MethodSummaryTable.jsx:22]: I'll leave the link decision up to @moneppo but would it make sense to just make the entire method name a link instead?

**Review by molly-moen — APPROVED** [review]
LGTM, one idea about the link

File: apps/src/templates/codeDocs/MethodSummaryTable.jsx
Comment (bethanyaconnor) [reply apps/src/templates/codeDocs/MethodSummaryTable.jsx:22]: I'm open to either! Unless I hear otherwise, I'll keep as-is for now and merge this when staging reopens so I don't forget about it.

**Pull request merged**
